addons/web/static/src/views/fields/properties/property_definition_selection.js

```diff
@@ -12,20 +13,55 @@ export class PropertyDefinitionSelection extends Component {
         // when we create a new option, it's added in the state
         // when we have finished to edit it (blur / enter) we propagate
         // the new value in the props
-        this.state = useState({ newOption: null });
+        this.state = useState({
+            newOption: null,
+            newOptionIndex: -1,
+        });
 
         this.propertyDefinitionSelectionRef = useRef("propertyDefinitionSelection");
         this.addButtonRef = useRef("addButton");
 
         useEffect(() => {
             // automatically give the focus to the new option if it is empty
+            if (this.state.newOptionIndex < 0) {
+                return;
+            }
             const inputs = this.propertyDefinitionSelectionRef.el.querySelectorAll(
                 ".o_field_property_selection_option input"
             );
-            if (inputs && inputs.length && !inputs[inputs.length - 1].value) {
-                inputs[inputs.length - 1].focus();
+            if (inputs && inputs.length && !inputs[this.state.newOptionIndex].value) {
+                inputs[this.state.newOptionIndex].focus();
             }
         });
+
+        useSortable({
+            enable: () => {
+                if (!this.props.canChangeDefinition || this.props.readonly) {
+                    return false;
+                }
+
+                const element = document.activeElement;
+                if (
+                    element &&
+                    element.tagName === "INPUT" &&
+                    element.parentNode.classList.contains("o_field_property_selection_option") &&
+                    (!element.value || !element.value.length)
+                ) {
+                    // try to move an empty option that will be removed
```

Comment:
Not sure what should be the expected behaviour here.
The comment suggests that we cannot move an empty option, but by looking at the condition above I would say that it makes it impossible to move any option if an empty option is focused (could not check on runbot it seems to be dead :/).

If it should prevent to move any option, so that the empty option is first removed, we could maybe use the `.o_field_property_selection_option input:placeholder-shown` selector to simplify the above condition ?

Reply:
> but by looking at the condition above I would say that it makes it impossible to move any option if an empty option is focused

Yes that's true, we loose the focus so the option is removed

> If it should prevent to move any option, so that the empty option is first removed, we could maybe use the .o_field_property_selection_option input:placeholder-shown selector to simplify the above condition ?

Sorry, I don't see how it can help, this trick is just to give the autofocus to the new option no ?

Reply:
ah ok i see :) yes let's keep it a bit more verbose / explicit

I updated the comment